packages/daemon/src/platforms/linux/linux.ts

```diff
@@ -0,0 +1,229 @@
+import {
+  execAsync,
+  getDirectories,
+  getFile,
+  logger,
+  pathExists,
+  saveFile,
+} from '@dfinity/http-proxy-core';
+import { Platform, PlatformProxyInfo } from '../typings';
+import { PlatformConfigs } from './typings';
+import { resolve } from 'path';
+import {
+  BASE_MOZILLA_PATH,
+  BASE_SNAP_MOZZILA_PATH,
+  CURL_RC_FILE,
+  FIREFOX_PROFILES_FOLDER,
+  MOZILLA_CERTIFICATES_FOLDER,
+  ROOT_CA_PATH,
+  findP11KitTrustPath,
+} from './utils';
+
+export class LinuxPlatform implements Platform {
+  constructor(
+    private readonly configs: PlatformConfigs,
+    private readonly username = String(process.env.LOGNAME ?? 'root')
+  ) {}
+
+  public async attach(): Promise<void> {
+    await this.setupDependencies();
+
+    logger.info(
+      `attaching proxy to system with: ` +
+        `host(${this.configs.proxy.host}:${this.configs.proxy.port}), ` +
+        `capath(${this.configs.ca.path}), ` +
+        `caname(${this.configs.ca.commonName})`
+    );
+
+    await this.trustCertificate(true, this.configs.ca.path);
+
+    await this.configureWebProxy(true, {
+      host: this.configs.proxy.host,
+      port: this.configs.proxy.port,
+    });
+  }
+
+  public async detach(): Promise<void> {
+    await this.setupDependencies();
+
+    logger.info(
+      `detaching proxy from system with: ` +
+        `host(${this.configs.proxy.host}:${this.configs.proxy.port}), ` +
+        `capath(${this.configs.ca.path}), ` +
+        `caname(${this.configs.ca.commonName})`
+    );
+
+    await this.trustCertificate(false, this.configs.ca.path);
+
+    await this.configureWebProxy(false, {
+      host: this.configs.proxy.host,
+      port: this.configs.proxy.port,
+    });
+  }
+
+  public async configureWebProxy(
+    enable: boolean,
+    { host, port }: PlatformProxyInfo
+  ): Promise<void> {
+    return new Promise<void>(async (ok, err) => {
+      try {
+        // configure proxy for curl
+        const curlrcPath = resolve(String(process.env.HOME), CURL_RC_FILE);
```

Comment:
Very minor and not necessary for this PR - but since it's the same as Mac, maybe we could consider a shared `configureCurlProxy()` function. Similarly for the Firefox policies.

Reply:
Agreed, i would leave that out of this PR as you suggested though.